### PR TITLE
Gsa 4 Play Feature - lazy loading

### LINKED ISSUE #4: F3(feat: Play)-Add lazy loading module
- Add lazy loading module for Play components

### PR DESCRIPTION
**Pull request by ChelleAmon**
## Changes
1. Implement lazy loading for Play Feature
2. Change routing navigation into HomeComponent template
3. Add game feature module into a play feature module

## Purpose
Make the Play Component/s load only if asked by the user.

## Approach
Because the application is small and the app redirects directly to home/play, we might not see the advantage of Lazy loading yet. But, for the sake of learning, I still want to go this route.

## Learning
Reference: mean-stack-apprentices/problox
https://www.freecodecamp.org/news/lazy-loading-in-angular-intro-to-ngmodules/

Closes #4

### CODE REVIEW

File: src/client/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
 import { LeaderboardComponent } from './components/leaderboard/leaderboard.component';
-import { PlayComponent } from './components/play/play.component';
 import { SignInComponent } from './components/sign-in/sign-in.component';
-import { GameComponent } from './pages/game/game.component';
 import { HomeComponent } from './pages/home/home.component';
 import { SignUpComponent } from './pages/sign-up/sign-up.component';
-import { GameInfoResolver } from './resolvers/game-info.resolver';
 
 const routes: Routes = [
-  {path: 'game', component: GameComponent, resolve: {gameInfo: GameInfoResolver}},
```
Comment (brycejjacks) [inline src/client/src/app/app-routing.module.ts:12]: again... keeping this line and maybe adding lazy loading to this path .. might be more of what we are looking for

**Review by brycejjacks — APPROVED** [review]
love what you have done here .. blame me for all the changes that have to be made.. i should have made things more clear, because you excelled in everything you set out to do. Thanks for cleaning up so many of those extra spaces lol...  Im realizing now that the conversations on the lines of code are blocking, so maybe just take note of them, resolve them, and then merge the branch yourself.. ill approve this pr ... sorry it took so long.

**Comment by brycejjacks** [comment]
just to reiterate in a simple message.... the nesting of the routes is not exactly what we are looking for ... im sure that we can lazy load both the play component and the game component, but the game page needs to be its own route... you can keep the lazy loading for the play if you desire .. we can lazy load everything im all for it, but the game module maybe shouldnt be in the play module ??? im sure it can be its own module ??? 

ps. triple question marks are not all the way thought out just pretty sure.. if you want to get on a call as you work on it to look at it together... ill think it through with you.

File: src/client/src/app/pages/home/home.component.html [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <ng-container >
 <nav>
     <ul>
-        <li><a routerLinkActive="active-link" routerLink="./play">Play</a></li>
+        <li><a routerLinkActive="active-link" routerLink="/home/play">Play</a></li>
```
[the remark this answers is not in the document]
Comment (ChelleAmon) [reply src/client/src/app/pages/home/home.component.html:4]: you are right lol! i changed this back to the original on my second PR

File: src/client/src/app/modules/play/components/play/play.component.ts [hunk 3]
```diff
@@ -30,8 +30,8 @@ export class PlayComponent implements OnInit {
   //   }
   }
 
-  playComputer() {   
-    this.router.navigate(['game'])
+  playComputer() {
+    this.router.navigate(['home/play/game'])
```
[the remark this answers is not in the document]
Comment (ChelleAmon) [reply src/client/src/app/modules/play/components/play/play.component.ts:34]: yes... will fix this on the next issue!

File: src/client/src/app/app-routing.module.ts (hunk 1 above)
Comment (ChelleAmon) [reply src/client/src/app/app-routing.module.ts:12]: will take care of it on next issue

**Comment by ChelleAmon** [comment]
merging branch. Going to fix the routes on next issue

**Pull request merged**